docs/specification.md

```diff
@@ -174,7 +180,18 @@ Everyone can make a new proposal, however, you have to freeze some tokens for th
 The proposer specifies how many frozen tokens they want to stake and this value is checked by
 the contract according to its compile-time configuration.
 
-This can only be performed in a proposing stage period, meaning one that's odd-numbered and
+For freezing the address should have corresponding amount of tokens of proper
+token type (token_id of storage.governance_token.token_id) in the governance
+FA2 contract. During 'Freeze' operation the BaseDAO contracts makes an FA2
+transfer on the governance contract to transfer tokens to its own address from
+the address who is freezing the tokens. Then it mints frozen tokens for the address.
+
+Unfreezing does it reverse, that is, the contract makes the FA2 transfer on governance
+contract to transfer tokens from its own address to the address that is doing unfreezing.
+It then burns the corresponding amount of tokens. Only frozen tokens that are not currently
+staked in a vote or proposal can be unfreezed.
```

Comment:
I think that this would be better in the `Stages` part just above, as it's not just about proposals.

Reply:
Yes, moved.